src/main/java/com/adyen/terminal/security/NexoCrypto.java

```diff
@@ -149,7 +151,7 @@ private void validateHmac(byte[] receivedHmac, byte[] decryptedMessage, NexoDeri
      */
     private byte[] generateRandomIvNonce() {
         byte[] ivNonce = new byte[NEXO_IV_LENGTH];
-        new Random().nextBytes(ivNonce);
+        SECURE_RANDOM.nextBytes(ivNonce);
```

Comment:
Hey @ykalayy,

We just double checked this with our security team, since we're not too comfortable with this kind of crypto stuff ourselves. They recommended us to use ```SecureRandom.getInstance("NativePRNGNonBlocking", "SUN")```, because this would presumably provide the securest PRG without blocking over all platforms. If you want to see all the various options for the SecureRandom class you can go over to this [site](https://docs.oracle.com/en/java/javase/11/docs/specs/security/standard-names.html#securerandom-number-generation-algorithms). Let me know your thoughts on this!

Best, Jilling
Adyen